src/tooling/piral-cli/src/apps/build-piral.ts

```diff
@@ -126,6 +131,14 @@ export async function buildPiral(baseDir = process.cwd(), options: BuildPiralOpt
       ignored,
     });
 
+    // run post piral build script if provided
+    if (postPiralBuildDevelop) {
```

Comment:
I think this block could be placed in a function.

```ts
function runLifecycle(root: string, script: string, type: string) {
...
}
```

where

```ts
runLifecycle(root, postPiralBuildDevelop, 'postPiralBuildDevelop');
```

It can be reused for the block below and maybe other blocks (pre), too.

Reply:
Agreed. Let me take another stab at it.